ui/ui-components/components/property/PropertySampleRecord.tsx

```diff
@@ -0,0 +1,87 @@
+import { useState, useCallback, useMemo } from "react";
+import { useDebouncedValue } from "../../hooks/useDebouncedValue";
+import { Actions, Models } from "../../utils/apiData";
+import SampleRecordCard, {
+  SampleRecordCardProps,
+} from "../record/SampleRecordCard";
+
+interface Props {
+  property: Models.PropertyType;
+  localFilters: Actions.PropertyView["property"]["filters"];
+}
+
+const PropertySampleRecord: React.FC<
+  Props & Omit<SampleRecordCardProps, "fetchRecord" | "highlightProperty">
+> = ({ property, localFilters, ...props }) => {
+  const { execApi } = props;
+  const debouncedProperty = useDebouncedValue(property, 1000);
+  const [errorMessage, setErrorMessage] = useState<string>();
+
+  const reloadKey = useMemo(() => {
+    return JSON.stringify([
+      debouncedProperty.id,
+      debouncedProperty.type,
+      debouncedProperty.unique,
+      debouncedProperty.isArray,
+      debouncedProperty.options,
+      debouncedProperty.filters,
+      JSON.stringify(localFilters),
+    ]);
+  }, [
+    debouncedProperty.id,
+    debouncedProperty.type,
+    debouncedProperty.unique,
+    debouncedProperty.isArray,
+    JSON.stringify(debouncedProperty.options),
+    JSON.stringify(debouncedProperty.filters),
+    JSON.stringify(localFilters),
+  ]);
+
+  const fetchRecord = useCallback(
+    async (recordId: string) => {
+      setErrorMessage(undefined);
+
+      const response: Actions.PropertyRecordPreview = await execApi(
+        "get",
+        `/property/${debouncedProperty.id}/recordPreview`,
+        {
+          options: debouncedProperty.options,
+          filters: localFilters,
+          recordId,
+        }
+      );
+
+      if (response?.record) {
+        setErrorMessage(
+          response.errorMessage
+            ? response.errorMessage.match(
+                /is required for a property of type/ // ignore errors about missing options
+              )
```

Comment:
I'll be this happens when you are filling out a new property and you haven't provided all the options yet.  Is there maybe a way not to hit the API until all the (required) options have a value in the UI?

Reply:
I looked into this and it's fairly challenging to get it right because we'd have to do some client-side validation that matches the form/submission. Besides, the rest of the sample record (other props, groups, dests) still get populated by this call so it's worth making the request.